src/lib/storage/up-storage.js

```diff
@@ -227,6 +227,57 @@ class ProxyStorage {
     headers[acceptEncoding] = headers[acceptEncoding] || 'gzip';
     // registry.npmjs.org will only return search result if user-agent include string 'npm'
     headers[userAgent] = headers[userAgent] || `npm (${this.userAgent})`;
+
+    if (!this.config.auth) {
+      return headers;
+    }
+
+    // copy headers to normalize keys
+    let copyHeaders = {};
+    Object.keys(headers).map((value) => {
```

Comment:
Two questions.

* Is this is necessary?
* Why not at the end?

Reply:
This to normalize name headers

```
input: Authorization, authOrization, authoriZaTion

to: authorization

```

Reply:
But you could di it below, right?

Reply:
Yes, but removed on commit below https://github.com/verdaccio/verdaccio/pull/332/commits/ab69b258ca15f32b76290a19e2b2175ab63bf1fd